### PR TITLE
Bump z3 version to 4.8.10

### PR DESCRIPTION
**Pull request by mmuesly**

### CODE REVIEW

**Comment by no-preserve-root** [comment]
This seems to work. The remaining compiler warnings are apparently caused by the generified Z3 API: https://github.com/Z3Prover/z3/issues/5033

**Comment by mmuesly** [comment]
Thank you @no-preserve-root for making the tests ready for future Z3 versions.

Due to the error in the z3 4.8.10 Java API for mkConcat expressions, I am slightly against releasing 4.8.10. Or we vote to due it, but then we have to patch the Java API before release due to this change: https://github.com/Z3Prover/z3/pull/4996/files
Do you have an opinion on this?

As we currently rely intensively on the Java API for String concatenation in DSE as well, I am against a partially broken release.

**Review by mmuesly — COMMENTED** [review]
LGTM, thank you for the back port! 👍

File: src/integrationTest/java/io/github/tudoaqua/z3turnkey/SolverInteractionTest.java [hunk 1]
```diff
@@ -31,55 +37,73 @@
...
+        Model m = solver.getModel();
+        Expr<BoolSort> evaluated = m.evaluate(b, true);
+        assertTrue(evaluated.isTrue(), "The model should evaluate");
+    }
+
+    /**
+     * Check that expression concatenation operates correctly.
+     */
+    @Test
+    public void testConcat() {
```
Comment (mmuesly) [inline src/integrationTest/java/io/github/tudoaqua/z3turnkey/SolverInteractionTest.java:97]: :)

**Pull request merged**